src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -1134,11 +1134,8 @@ private void ComputeNullableValueOffset()
 
         protected virtual void ComputeValueTypeFieldPadding()
         {
-            // All objects that can have appreciable which can be derived from size compute ValueTypeFieldPadding.
-            // Unfortunately, the name ValueTypeFieldPadding is now wrong to avoid integration conflicts.
-
-            // Interfaces, sealed types, and non-DefTypes cannot be derived from
-            if (_type.IsInterface || !_type.IsDefType || (_type.IsSealed() && !_type.IsValueType))
+            // Only valuetypes need to compute the padding.
```

Comment:
This will also be a small size saving - because of universal shared code, most MethodTables had rare flags with this value (making rare flags not so rare).